libs/langchain/langchain/agents/tools_agent/base.py

```diff
@@ -0,0 +1,96 @@
+from typing import Sequence
+
+from langchain_core.language_models import BaseLanguageModel
+from langchain_core.prompts.chat import ChatPromptTemplate
+from langchain_core.runnables import Runnable, RunnablePassthrough
+from langchain_core.tools import BaseTool
+
+from langchain.agents.format_scratchpad.tools import (
+    format_to_tool_messages,
+)
+from langchain.agents.output_parsers.tools import ToolsAgentOutputParser
+
+
+def create_tools_agent(
+    llm: BaseLanguageModel, tools: Sequence[BaseTool], prompt: ChatPromptTemplate
```

Comment:
`BaseLanguageModelLike` would be better to account for `.bind()`

Reply:
Oh wait, never mind you wouldn't be able to check for the presence of `bind_tools()` below